SSA/Experimental/ASTPrettyPrinter.lean

```diff
@@ -0,0 +1,278 @@
+import Lean
+import SSA.Core.MLIRSyntax.AST
+import SSA.Core.MLIRSyntax.GenericParser
+import SSA.Projects.InstCombine.LLVM.PrettyEDSL
+open Lean PrettyPrinter Delaborator SubExpr MLIR AST Elab Term Syntax
+open PrettyPrinter
+
+-- AFFINE SYNTAX
+-- ============
+
+@[app_unexpander AST.AffineExpr.Var]
+def unexpandAffineExprVar : Unexpander
+  | `($_ $xstr:str) =>
+    let xraw := mkIdent $ Name.mkSimple xstr.getString
+    `([affine_expr| $xraw:ident])
+  | _ => throw ()
+
+
+@[app_unexpander AST.AffineTuple.mk]
+def unexpandAffineTuplemk : Unexpander
+  | `($_ [$[$terms],*]) =>
+     let affexprs : Array (TSyntax `affine_expr) := terms.map fun term => match term with
+      | `([affine_expr| $n]) => n
+      | _ => panic! "affine_expr is illformed"
+    `([affine_tuple|($affexprs,*)])
+  | _ => throw ()
+
+
+@[app_unexpander AST.AffineMap.mk]
+def unexpandAffineMapmk : Unexpander
+  | `($_ [affine_tuple|($xs,*)] [affine_tuple|($ys,*)]) =>
+    `([affine_map|affine_map< ($xs,*) -> ($ys,*)>])
+  | _ => throw ()
+/-- info: [affine_expr|foo] : AffineExpr -/
+#guard_msgs in #check [affine_expr|foo]
+
+/-- info: [affine_tuple|(a,b,c)] : AffineTuple -/
+#guard_msgs in #check [affine_tuple| (a,b,c) ]
+
+/-- info: [affine_map|affine_map<(a,b)->(c,d)>] : AffineMap -/
+#guard_msgs in #check [affine_map|affine_map<(a,b)->(c,d)>]
+
+
+-- EDSL OPERANDS
+-- ==============
+-- TODO?: unexpander for [mlir_op_operand | $$($q)]
+
+
+-- TODO: this should become: [mlir_op_operand| %x] : MLIR.SSAVal
+/--
+info: SSAVal.name (EDSL.IntToString 0) : SSAVal
+-/
+#guard_msgs in #check [mlir_op_operand| %0]
+
+@[app_unexpander AST.SSAVal] -- broken?
```

Comment:
This was previously `@[app_unexpander AST.SSAVal.SSAVal]` and did not compile. While it now compiles, the output is not as expected.